strawberry/fastapi/__init__.py

```diff
@@ -0,0 +1,188 @@
+import json
+from datetime import timedelta
+from typing import Any, Optional
+
+from fastapi import (
+    APIRouter,
+    BackgroundTasks,
+    Depends,
+    Request,
+    Response,
+    WebSocket,
+    status,
+)
+from fastapi.responses import HTMLResponse, JSONResponse, PlainTextResponse
+from strawberry.asgi.utils import get_graphiql_html
+from strawberry.exceptions import MissingQueryError
+from strawberry.fastapi.handlers.graphql_transport_ws_handler import (
+    GraphQLTransportWSHandler,
+)
+from strawberry.fastapi.handlers.graphql_ws_handler import GraphQLWSHandler
+from strawberry.file_uploads.utils import replace_placeholders_with_files
+from strawberry.http import GraphQLHTTPResponse, parse_request_data, process_result
+from strawberry.schema import BaseSchema
+from strawberry.subscriptions import GRAPHQL_TRANSPORT_WS_PROTOCOL, GRAPHQL_WS_PROTOCOL
+from strawberry.types import ExecutionResult
+from strawberry.utils.debug import pretty_print_graphql_operation
+
+
+async def get_root_value():
+    return None
+
+
+async def get_context(
+    background_tasks: BackgroundTasks, request: Request = None, ws: WebSocket = None
+) -> Optional[Any]:
+    return {"request": request or ws, "background_tasks": background_tasks}
+
+
+class GraphQL(APIRouter):
+    graphql_ws_handler_class = GraphQLWSHandler
+    graphql_transport_ws_handler_class = GraphQLTransportWSHandler
+
+    def __init__(
+        self,
+        schema: BaseSchema,
+        prefix: str = "",
+        graphiql: bool = True,
+        keep_alive: bool = False,
+        keep_alive_interval: float = 1,
+        debug: bool = False,
+        root_value_getter=get_root_value,
+        context_getter=get_context,
+        subscription_protocols=(GRAPHQL_TRANSPORT_WS_PROTOCOL, GRAPHQL_WS_PROTOCOL),
+        connection_init_wait_timeout: timedelta = timedelta(minutes=1),
+    ):
+        super().__init__(prefix=prefix)
```

Comment:
Not sure if [`redirect_slashes`](https://github.com/encode/starlette/blob/28088573aebc70807ef2aa422000cddb93b57a70/starlette/routing.py#L527-L541) makes sense as well? 
```suggestion
        connection_init_wait_timeout: timedelta = timedelta(minutes=1),
        default: Optional[ASGIApp] = None,
        on_startup: Optional[Sequence[Callable[[], Any]]] = None,
        on_shutdown: Optional[Sequence[Callable[[], Any]]] = None,
    ):
        super().__init__(
            prefix=prefix,
            default=default,
            on_startup=on_startup,
            on_shutdown=on_shutdown
        )
```